src/Diagnostics/atmos_refstate_perturbations.jl

```diff
@@ -0,0 +1,268 @@
+# AtmosRefStatePerturbations
+#
+# Computes perturbations from the reference state and outputs them
+# on the specified interpolated grid.
+
+using ..Atmos
+using ..Mesh.Topologies
+using ..Mesh.Grids
+using ..Thermodynamics
+
+function vars_atmos_refstate_perturbations(m::AtmosModel, FT)
+    @vars begin
+        ref_state::vars_atmos_refstate_perturbations(m.ref_state, FT)
+    end
+end
+vars_atmos_refstate_perturbations(::ReferenceState, FT) = @vars()
+function vars_atmos_refstate_perturbations(rs::HydrostaticState, FT)
+    @vars begin
+        rho::FT
+        pres::FT
+        temp::FT
+        et::FT
+        qt::FT
+    end
+end
+num_atmos_refstate_perturbation_vars(m, FT) =
+    varsize(vars_atmos_refstate_perturbations(m, FT))
+atmos_refstate_perturbation_vars(m, array) =
+    Vars{vars_atmos_refstate_perturbations(m, eltype(array))}(array)
+
+function atmos_refstate_perturbations!(
+    atmos::AtmosModel,
+    state,
+    aux,
+    thermo,
+    vars,
+)
+    atmos_refstate_perturbations!(
+        atmos.ref_state,
+        atmos,
+        state,
+        aux,
+        thermo,
+        vars,
+    )
+    return nothing
+end
+function atmos_refstate_perturbations!(
+    ::ReferenceState,
+    ::AtmosModel,
+    state,
+    aux,
+    thermo,
+    vars,
+)
+    return nothing
+end
+function atmos_refstate_perturbations!(
+    rs::HydrostaticState,
+    atmos::AtmosModel,
+    state,
+    aux,
+    thermo,
+    vars,
+)
+    vars.ref_state.rho = state.ρ - aux.ref_state.ρ
+    vars.ref_state.pres = thermo.pres - aux.ref_state.p
+    vars.ref_state.temp = thermo.temp - aux.ref_state.T
+    vars.ref_state.et =
+        (state.ρe / state.ρ) - (aux.ref_state.ρe / aux.ref_state.ρ)
+    # FIXME properly
+    if atmos.moisture isa EquilMoist
+        vars.ref_state.qt =
+            (thermo.moisture.ρq_tot / state.ρ) -
+            (aux.ref_state.ρq_tot / aux.ref_state.ρ)
+    else
+        vars.ref_state.qt = Inf
```

Comment:
Maybe I'm just misunderstanding here, but if `qt` is the total moisture, shouldn't this zero for a dry model? Setting it to `Inf` here seems unsafe.

Reply:
The standard `missing` value in NetCDF is `NaN` not `Inf` 
(per [this](https://www.unidata.ucar.edu/software/netcdf/workshops/2011/bestpractices/MissingData.html)), I'll fix this thanks.